Robus/src/msgAlloc.c

```diff
@@ -65,71 +155,87 @@ static void MsgAlloc_ClearMemSpace(uint8_t *from, uint8_t *to)
  ******************************************************************************/
 void MsgAlloc_Init(void)
 {
-    /******** Init **********
-     * - init current_msg to msg_buffer
-     * - init data_ptr to msg_buffer
-     */
+    //******** Init global vars pointers **********
+    current_msg = (msg_t *)msg_buffer;
+    data_ptr = (uint8_t *)msg_buffer;
+    module_concerned_stack_pointer = 0;
 }
 
 /******************************************************************************
  * @brief Invalid the current message header by removing it (data will be ignored).
  * @param None
  * @return None
  ******************************************************************************/
-void MsgAlloc_InvalidHeader(void)
+void MsgAlloc_InvalidMsg(void)
 {
-    /******** Remove the header  *********
-     * - move data_ptr to current_msg
-     */
+    //******** Remove the header by reseting data_ptr *********
+    data_ptr = (uint8_t *)current_msg;
 }
 
 /******************************************************************************
  * @brief Valid the current message header by preparing the allocator to get the message data
- * @param None
+ * @param concerned_module : The module concerned by this message
  * @return None
  ******************************************************************************/
-void MsgAlloc_ValidHeader(void)
+void MsgAlloc_ValidHeader(vm_t *concerned_module)
 {
-    /******** Prepare the allocator to get data  *********
-     * - Be sure that the end of msg_buffer is after data_ptr + header_t.size
-     * - If the end of buffer is before the end of the message :
-     *     - clean space between &msg_buffer[0] and (&msg_buffer[0] + sizeof(header_t) + header_t.size) using MsgAlloc_ClearMemSpace(uint8_t *from, uint8_t *to)
-     *     - Copy the header at the begining of msg_buffer
-     *     - move current_msg to &msg_buffer[0]
-     *     - move data_ptr to current_msg + sizeof(header_t)
-     * - If not 
-     *     - clean space between data_ptr and (data_ptr + header_t.size) using MsgAlloc_ClearMemSpace(uint8_t *from, uint8_t *to)
-     */
+    //******** Prepare the allocator to get data  *********
+    // Save the concerned module pointer into the concerned module pointer stack
+    module_concerned_by_current_msg[module_concerned_stack_pointer++] = concerned_module;
+    // Be sure that the end of msg_buffer is after data_ptr + header_t.size
+    if (MsgAlloc_ClearMemSpace((void *)data_ptr, (void *)(data_ptr + current_msg->header.size + 2)) == FAIL)
```

Comment:
This function could be call multiple times for the same message. We should do the MsgAlloc_ClearMemSpace only once.